ocaml/typing/includecore.ml

```diff
@@ -360,6 +361,10 @@ let report_record_mismatch first second decl env ppf err =
       pr "@[<hv>Their internal representations differ:@ %s %s %s.@]"
         (choose ord first second) decl
         "uses unboxed float representation"
+  | Ufloat_representation ord ->
+      pr "@[<hv>Their internal representations differ:@ %s %s %s.@]"
+        (choose ord first second) decl
+        "uses float# representation"
```

Comment:
After staring at this for a minute, I think it's unreachable, because you'd always hit a layout error first.  But I think it might be reachable in the future when we allow any in more spots, with an example like:
```ocaml
module S : sig
  type t : any
  type u = { f1 : t; f2 : t }
end = struct
  type t = float#
  type u = { f1 : t; f2 : t }
end;;
```
I've added a CR.